File: src/io/github/talkarcabbage/rstimer/fxgui/ProgressAnimTimer.java
package io.github.talkarcabbage.rstimer.fxgui;

import com.google.common.collect.BiMap;
import com.minepop.talkar.util.ConfigManager;

import io.github.talkarcabbage.rstimer.FXController;
import io.github.talkarcabbage.rstimer.Timer;
import javafx.animation.AnimationTimer;

/**
 * 
 * @author Talkarcabbage
 *
 */
public class ProgressAnimTimer extends AnimationTimer {

	final BiMap<ProgressPane, Timer> map;
	int animCycle;
	
	public ProgressAnimTimer(BiMap<ProgressPane, Timer> biMap) {
		map = biMap;
		animCycle = 1000000;
	}

	@Override
	public void handle(long now) {
		if (++animCycle >= ConfigManager.getInstance().getFramesPerUpdate()) {
			animCycle = 0;
			map.forEach( (pane, timer) -> {
				pane.setProgress(timer.getPercentageComplete());
				pane.bar.getTooltip().setText(FXController.formatTime(timer.getTimeRemaining()));
				pane.label.getTooltip().setText(FXController.formatTime(timer.getTimeRemaining()));
			});	
		}
	}
}

File: src/main/kotlin/io/github/talkarcabbage/rstimer/fxgui/ProgressAnimTimer.kt
package io.github.talkarcabbage.rstimer.fxgui

import com.google.common.collect.BiMap

import io.github.talkarcabbage.rstimer.FXController
import io.github.talkarcabbage.rstimer.Timer
import io.github.talkarcabbage.rstimer.persistence.ConfigManager
import javafx.animation.AnimationTimer

/**
 *
 * @author Talkarcabbage
 */
class ProgressAnimTimer(internal val map: BiMap<ProgressPane, Timer>) : AnimationTimer() {
	internal var animCycle: Int = 0

	init {
		animCycle = 1000000
	}

	override fun handle(now: Long) {
		if (++animCycle >= ConfigManager.framesPerUpdate) {
			animCycle = 0
			map.forEach { (pane, timer) ->
				pane.setProgress(timer.percentageComplete.toLong())
				pane.bar.getTooltip().setText(FXController.formatTime(timer.timeRemaining))
				pane.labelObject.getTooltip().setText(FXController.formatTime(timer.timeRemaining))
			}
		}
	}
}